### PR TITLE
Docker images for v2

### PR DESCRIPTION
**Pull request by JacksonBurns**
This PR:
 - add automatic building of PyPI and Docker images back to the CI, which we had forgotten to add during the initial release of v2
 - updates the usage docs to be more helpful and concise
 - updates the dockerfile to avoid installing gpu pytorch when the container only runs cpu

I have already built and pushed a v2.0.0 image manually to Dockerhub. In the future we will have builds continuously made - when we want to do a release we just pull the built image, tag it, and push it back.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (kevingreenman) [inline .github/workflows/ci.yml:144]: why is this being removed?

File: .github/workflows/ci.yml
Comment (kevingreenman) [inline .github/workflows/ci.yml:140]: since we've added this here, do we still need the [`publish.yml`](https://github.com/chemprop/chemprop/blob/main/.github/workflows/publish.yml) file?

**Review by kevingreenman — COMMENTED** [review]
@JacksonBurns thanks for the PR! I just have a couple questions before we merge

File: .github/workflows/ci.yml
Comment (JacksonBurns) [reply .github/workflows/ci.yml:144]: This was in place to avoid having to manually specify the "run tests" job for each platform and Python version when creating branch protection rules, which I just ended up doing anyway making this unnecessary

File: .github/workflows/ci.yml
Comment (JacksonBurns) [reply .github/workflows/ci.yml:140]: We do not, but we could also delete this and keep the publish file instead if we want. Your call

Comment (kevingreenman) [reply .github/workflows/ci.yml:140]: doesn't matter to me. I suggest deleting the publish file since that looks more outdated. Should be good to merge after removing that

Comment (JacksonBurns) [reply .github/workflows/ci.yml:140]: Done

**Comment by JacksonBurns** [comment]
ugh.

the latest macos runners (the apple silicon ones) dont support the default DDP strategy in lightning _and_ are not accessible from our runners which causes errors when lightning attempts to use it when `accelerators="auto"`.

@kevingreenman I guess we should resolve the test failures in a separate PR before this one.

**Comment by kevingreenman** [comment]
did this just change today? I thought all the tests were passing earlier today

**Comment by JacksonBurns** [comment]
Yeah this only just changed, the CI is failing now too: https://github.com/chemprop/chemprop/actions/runs/8845256777

**Comment by JacksonBurns** [comment]
It was probably torch 2.3, which just came out: https://pytorch.org/blog/pytorch2-3/

**Comment by JacksonBurns** [comment]
finally found someone else reporting the error: https://discuss.pytorch.org/t/mps-back-end-out-of-memory-on-github-action/189773/2

pytorch knows about it and has closed the issue tracking it: https://github.com/pytorch/pytorch/issues/111449#issuecomment-1767613883

and yet this issue remains open showing that this is not resolved: https://github.com/actions/runner-images/issues/9254

solution is to not run on the latest macos... sad

**Comment by JacksonBurns** [comment]
@kevingreenman switching back to macos13 which uses intel chips works. If you are ok with this we can merge as-is to get CI passing and re-open #776 to track adding apple silicon runners back

**Review by kevingreenman — APPROVED** [review]

**Comment by kevingreenman** [comment]
@JacksonBurns that sounds good to me. It looks like the status checks are still showing the `macos-latest` tests as being expected/required, but if I click on "Details" to see the more in-depth status of any of the tests, it shows that everything is complete, and `macos-latest` is not present on that page. Are these still showing up on this PR because they were required when the PR was opened (or maybe because they're required in the target branch)? If so, I think we can use the "bypass branch protections" option.

**Comment by JacksonBurns** [comment]
yes, the branch protection still expects macos latest. With your approval here, I have now changed this rule to instead look for `macos13` and you should be able to merge this normally.

**Comment by kevingreenman** [comment]
thanks!

**Pull request merged**
